hal/armv7m/imxrt/pmap.c

```diff
@@ -71,109 +56,20 @@ addr_t pmap_getMaxVAdrr(void)
 /* Function creates empty page table */
 int pmap_create(pmap_t *pmap, pmap_t *kpmap, page_t *p, void *vaddr)
 {
-	return EOK;
-}
-
-
-int pmap_getMapParameters(u8 id, void **start, void **end)
-{
-	int i;
-
-	/* Stop reading parameters */
-	if (id >= syspage->mapssz)
-		return EOK;
-
-	*start = (void *)syspage->maps[id].start;
-	*end = (void *)syspage->maps[id].end;
-
-	if (*end <= *start || ((addr_t)*start & (SIZE_PAGE - 1)) || ((addr_t)*end & (SIZE_PAGE - 1)))
-		return -EINVAL;
-
-	/* Check if new map overlap with existing one */
-	for (i = 0; i < syspage->mapssz; ++i) {
-		if (i == id)
-			continue;
-
-		if ((*start < (void *)syspage->maps[i].end) && (*end > (void *)syspage->maps[i].start))
-			return -EINVAL;
-	}
-
-	/* Continue reading map parameters */
-	return 1;
-}
-
-
-static void pmap_getMinOverlappedRange(void *memStart, void *memStop, void **segStart, void **segStop, void **minSegStart, void **minSegStop)
-{
-	if ((memStart < *segStop) && (memStop > *segStart)) {
-		if (memStart > *segStart)
-			*segStart = memStart;
-
-		if (memStop < *segStop)
-			*segStop = memStop;
-
-		if (*segStart < *minSegStart) {
-			*minSegStart = *segStart;
-			*minSegStop = *segStop;
-		}
-	}
-}
-
-
-void pmap_getAllocatedSegment(void *memStart, void *memStop, void **estart, void **estop)
-{
-	int i;
-	void *segStart = 0, *segStop = 0;
-	void *minSegStart, *minSegStop;
-
-	minSegStart = (void *)-1;
-	minSegStop = 0;
-
-	/* Check syspage segment */
-	segStart = syspage;
-	segStop = ((void *)syspage + syspage->syspagesz);
-	pmap_getMinOverlappedRange(memStart, memStop, &segStart, &segStop, &minSegStart, &minSegStop);
-
-	/* Check kernel's .text segment */
-	segStart = (void *)(syspage->kernel.text);
-	segStop = ((void *)syspage->kernel.text + syspage->kernel.textsz);
-	pmap_getMinOverlappedRange(memStart, memStop, &segStart, &segStop, &minSegStart, &minSegStop);
-
-	/* Check kernel's .data segment */
-	segStart = (void *)syspage->kernel.data;
-	segStop = ((void *)syspage->kernel.data + syspage->kernel.datasz);
-	pmap_getMinOverlappedRange(memStart, memStop, &segStart, &segStop, &minSegStart, &minSegStop);
-
-	/* Check programs' segments */
-	for (i = 0; i < syspage->progssz; ++i) {
-		segStop = (void *)syspage->progs[i].end;
-		segStart = (void *)syspage->progs[i].start;
-		pmap_getMinOverlappedRange(memStart, memStop, &segStart, &segStop, &minSegStart, &minSegStop);
-	}
-
-	if (minSegStart != (void *)-1) {
-		*estart = minSegStart;
-		*estop = minSegStop;
-	}
-}
-
-
-int pmap_getMapsCnt(void)
-{
-	return syspage->mapssz;
+	return 0;
 }
 
 
 void _pmap_init(pmap_t *pmap, void **vstart, void **vend)
 {
-	(*vstart) = (void *)(((addr_t)(syspage->kernel.bss + syspage->kernel.bsssz + 1024 + 256) + 7) & ~7);
+	addr_t size = (addr_t)&_end - (addr_t)&__bss_start;
+
+	(*vstart) = (void *)(((addr_t)((addr_t)&__bss_start + size + 1024 + 256) + 7) & ~7);
```

Comment:
Hmm, we can fetch initial stack size from a first vector in the vector table.

Reply:
Done